### ISSUE TITLE
Adding a type to an optional parameter marked by a question token puts the question token in the wrong spot

### ISSUE THREAD

**Issue by gregjacobs** [dsherret/ts-simple-ast#373]
Hey @dsherret. Looks like I found another interesting one. Trying to add a type to an optional parameter marked by a question token seems to put the '?' token in the wrong place in the output code. Check out this reproduction:

```ts
import Project, { MethodDeclaration, SyntaxKind } from 'ts-simple-ast';

const project = new Project();
const sourceFile = project.createSourceFile( 'my-file.ts', `
	class MyClass {
		someMethod( param1? ) {}
	}
` );


const method = sourceFile
	.getDescendantsOfKind( SyntaxKind.MethodDeclaration )[ 0 ] as MethodDeclaration;

const param1 = method.getParameters()[ 0 ];
param1.setType( 'any' );

console.log( 'Transformed:' );
console.log( sourceFile.getFullText() );
```

Basically the original source code:

```ts
class MyClass {
    someMethod( param1? ) {}
}
```

Gets changed into:

```ts
class MyClass {
    someMethod( param1: any? ) {}
}
```

Where the correct output should be:

```ts
class MyClass {
    someMethod( param1?: any ) {}
}
```

Best,
Greg

**Comment by dsherret** [comment]
As always, thanks for the excellent report @gregjacobs! Luckily for you I happened to be programming tonight so this is fixed now in 12.7.1.

**Comment by gregjacobs** [comment]
@dsherret You're the man! Thanks so much for the fix, and so quickly! Much appreciated!

Best,
Greg
